backend/typescript/services/implementations/campService.ts

```diff
@@ -1,17 +1,93 @@
-import { CreateCampDTO, CamperCSVInfoDTO, CampDTO } from "../../types";
+import {
+  CreateCampDTO,
+  CampDTO,
+  CamperCSVInfoDTO,
+  GetCampDTO,
+  FormQuestionDTO,
+} from "../../types";
 import ICampService from "../interfaces/campService";
-import MgCampSession, { CampSession } from "../../models/campSession.model";
-import MgCamper, { Camper } from "../../models/camper.model";
 import { getErrorMessage } from "../../utilities/errorUtils";
 import { generateCSV } from "../../utilities/CSVUtils";
 import logger from "../../utilities/logger";
 import MgCamp, { Camp } from "../../models/camp.model";
-import MgFormQuestion from "../../models/formQuestion.model";
+import MgCampSession, { CampSession } from "../../models/campSession.model";
+import MgFormQuestion, { FormQuestion } from "../../models/formQuestion.model";
+import MgCamper, { Camper } from "../../models/camper.model";
 
 const Logger = logger(__filename);
 
 class CampService implements ICampService {
   /* eslint-disable class-methods-use-this */
+  async getCamps(): Promise<GetCampDTO[]> {
+    try {
+      const camps: Camp[] | null = await MgCamp.find({})
+        .populate({
+          path: "campSessions",
+          model: MgCampSession,
+          populate: {
+            path: "campers",
+            model: MgCamper,
+          },
+        })
+        .populate({
+          path: "formQuestions",
+          model: MgFormQuestion,
+        });
+
+      if (!camps) {
+        return [];
+      }
+
+      return camps.map((camp) => {
+        const formQuestionArr = camp.formQuestions as FormQuestion[];
+        const formQuestions = formQuestionArr.map(
+          (formQuestion: FormQuestion) => {
+            const {
+              id,
+              type,
+              question,
+              required,
+              description,
+              options,
+            } = formQuestion;
+            let result: FormQuestionDTO = { id, type, question, required };
+            if (description) {
+              result = { ...result, description };
+            }
+            if (options) {
+              result = { ...result, options };
+            }
+
+            return result;
+          },
+        );
+        const campSessionsArr = camp.campSessions as CampSession[];
+        const campSessions = campSessionsArr.map((campSession) => ({
+          dates: campSession.dates.map((date) => date.toString()),
+          startTime: campSession.startTime,
+          endTime: campSession.endTime,
+          active: campSession.active,
+        }));
+
+        return {
+          id: camp.id,
+          ageLower: camp.ageLower,
+          ageUpper: camp.ageUpper,
+          capacity: camp.capacity,
+          name: camp.name,
+          description: camp.description,
+          location: camp.location,
+          fee: camp.fee,
+          formQuestions,
+          campSessions,
```

Comment:
can we actually update this to return campers and waitlist fields for each campSession, but just the numbers of people on each list? so the frontend will have the number of people on the campers and waitlist - the designs require thiss https://www.figma.com/file/uRMwPpIaSgsMGdbJzZziMA/FON-%2F-Management-Experience?node-id=407%3A7960

Reply:
sure, added! In design dock, I see that they have referred to it as "registrations"  and "waitlist", so the new field include "registrations" and "waitlists" containing number of campers in each of them.